core/src/test/java/org/ehcache/EhcacheBasicPutEventsTests.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache;
+
+import org.ehcache.event.CacheEvent;
+import org.ehcache.event.CacheEventListener;
+import org.ehcache.events.CacheEventNotificationService;
+import org.ehcache.events.CacheEventNotificationServiceImpl;
+import org.ehcache.exceptions.CacheAccessException;
+import org.ehcache.exceptions.CacheWritingException;
+import org.ehcache.spi.loaderwriter.CacheLoaderWriter;
+import org.ehcache.statistics.CacheOperationOutcomes;
+import org.ehcache.util.EhcacheBasicTestUtil;
+import org.ehcache.util.IsCreated;
+import org.ehcache.util.IsUpdated;
+import org.junit.Test;
+import org.mockito.InOrder;
+import org.mockito.Mock;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.argThat;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.inOrder;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+
+
+/**
+ * @author
+ */
+@SuppressWarnings("ThrowableResultOfMethodCallIgnored")
+public class EhcacheBasicPutEventsTests extends EhcacheBasicCrudBase{
+
+    @Mock
+    protected CacheLoaderWriter<String, String> cacheLoaderWriter;
+
+
+    @Mock
+    protected CacheEventListener testCacheEventListener;
+
+    protected CacheEventNotificationService cacheEventNotificationService;
+
+    protected IsCreated isCreated = new IsCreated();
+
+    protected IsUpdated isUpdated = new IsUpdated();
+
```

Comment:
I guess it would be interesting to add some CacheLoaderWriter failing test case too